python/mxnet/io.py

```diff
@@ -515,17 +515,10 @@ def _init_data(data, allow_empty, default_name):
     return list(data.items())
 
 def _has_instance(data, dtype):
-    """return True if data has instance of dtype"""
-    if isinstance(data, dtype):
-        return True
-    if isinstance(data, list):
-        for v in data:
-            if isinstance(v, dtype):
-                return True
-    if isinstance(data, dict):
-        for v in data.values():
-            if isinstance(v, dtype):
-                return True
+    """return True if data has instance of dtype after _init_data"""
+    for item in data:
+        if isinstance(item[1], dtype):
```

Comment:
why `item[1] `?

Reply:
This function is called after _init_data which transforms the data to [ ( name, ndarray), ... ] format.

Reply:
I see. Do you mind updating the comment to note the format of the input? It's not obvious unless someone read the code for _init_data

Reply:
One option to improve readability:
```
name, arr = item[1]
if isinstance(...):
```